### ISSUE TITLE
[Feature] Implement Authorization Header Token check

### ISSUE THREAD

**Issue by pandevim** [YouTubeNLP/Backend#22]
Currently anyone can use our api endpoints as they're isn't any type of authentication or credential check as we receive request from client. 

The API endpoints for all services (excluding root) should implement some kind of check for the Header `x-auth-key` in the HTTP Request, to give a 200 (OK) status code for correct token, if not, we will respond with a 401 (Unauthorized) for incorrect auth token.

---
This issue is blocked with #21

**Comment by karkipra** [comment]
https://medium.com/data-rebels/fastapi-authentication-revisited-enabling-api-key-authentication-122dc5975680 this seems to align with our interests

**Comment by YashSahsani** [comment]
@karkipra we can use that but the API key will be hardcoded the API so I was thinking like we would create a jwt token dynamically at runtime when a user logs in, fronted would make a call to an endpoint like /get/token than we would generate a jwt token which would have payload like username,time-of-generation-of-token,expiration-time-of-token and stuff. It would be better because it would have encryption and it would good while logging user's activities.

**Comment by karkipra** [comment]
> @karkipra we can use that but the API key will be hardcoded so I was thinking like we would create a jwt token dynamically at runtime when a user logs in, fronted would make a call to an endpoint like /get/token than we would generate a jwt token which would have payload like username,time-of-generation-of-token,expiration-time-of-token and stuff. It would be better because it would have encryption and it would be good while logging user's activities.

Seems like a good idea. Could this also double for user login authentication @YashSahsani?

**Comment by edward2018211** [comment]
Yep, I think Yash's idea sounds good

**Comment by YashSahsani** [comment]
@karkipra yeah we can even use that for user login authentication to but mainly I have seen it used in API calls; for user auth we can use like session_id or something like that. Also, jwt token was just for explanation there are many different alternatives that we can use like bearer authentication header or oauth2. For better understanding you can take a look at https://jwt.io/.
